testing/runner.py

```diff
@@ -209,6 +211,10 @@ def on_error(self):
             self.screenshot()
             raise
 
+    def __del__(self):
+       self.shutdown_called = True
```

Comment:
overriding `__del__` in python is almost never correct fwiw

Reply:
good point, thanks for letting me know.  I'll improve that soon.